tests/unit/configFile.spec.js

```diff
@@ -133,4 +133,239 @@ describe("configFile.js", () => {
+        configurationURL:
+          "https://raw.githubusercontent.com/AdvancedThreatAnalytics/threat-analytics-search/main/settings.json",
+        useGroups: true,
+        configEncrypted: false,
+        configEncryptionKey: "",
+        autoUpdateConfig: false,
+      };
+
+      // Shouldn't override if the flag is false.
+      await ConfigFile.default.parseJSONFile(defaultSettings, false);
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings).toEqual({});
+
+      // Should override if the flag is true.
+      await ConfigFile.default.parseJSONFile(defaultSettings, true);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings).toEqual(expect.objectContaining(expectedParsedData));
+    });
+
+    it("groups name should be updated only if the mergeGroups is true", async () => {
+      var newData = _.cloneDeep(defaultSettings);
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      const defaultProvidersGroup = [
+        { name: "IP Lookup", enabled: true },
+        { name: "Domain", enabled: true },
+        { name: "Hash", enabled: false },
+      ];
+
+      // If mergeGroups is false shouldn't override.
+      settings.mergeGroups = false;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.parseJSONFile(newData);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings.providersGroups).toEqual(defaultProvidersGroup);
+
+      // If mergeGroups is false should override providersGroups.
+      defaultProvidersGroup[2].name = "New Hash";
+      newData.groups[2][1] = "New Hash";
+      settings.mergeGroups = true;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.parseJSONFile(newData);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings.providersGroups).toEqual(defaultProvidersGroup);
+    });
+
+    it("search providers are merged, overridden and ignored depending on corresponding flag", async () => {
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      var serachProviders = await LocalStore.getOne(StoreKey.SEARCH_PROVIDERS);
+
+      // By default flag should be merge and duplicate values shouldn't be added.
+      var newData = {
+        searchproviders: [
+          [
+            -1,
+            "Google new",
+            "http://www.google2.com/search?q=TESTSEARCH",
+            true,
+            true,
+            3,
+            false,
+            "",
+            false,
+            "",
+          ],
+          [
+            -1,
+            "D - AlienVault OTX Domain",
+            "https://otx.alienvault.com/indicator/domain/TESTSEARCH",
+            true,
+            true,
+            2,
+            false,
+            "",
+            false,
+            "",
+          ],
+        ],
+      };
```

Comment:
Here could be more easy to just do `const newData = defaultSettings.providers.splice(0,2);`